processor/src/main/java/org/mapstruct/ap/internal/model/ForgedMethod.java

```diff
@@ -369,4 +377,50 @@ public int hashCode() {
         result = 31 * result + ( returnType != null ? returnType.hashCode() : 0 );
         return result;
     }
+
+    /**
+     * Not all options are relevant for forged methods.
+     */
+    public static class WrappedMappingMethodOptions extends MappingMethodOptions {
+
+        public WrappedMappingMethodOptions(MappingMethodOptions options) {
+            super(
+                options.getMapper(),
+                options.getMappings(),
+                options.getIterableMapping(),
+                options.getMapMapping(),
+                new WrappedBeanMappingOptions( options.getBeanMapping() ),
+                options.getEnumMappingOptions(),
+                options.getValueMappings()
+            );
+        }
+    }
+
+    public static class WrappedBeanMappingOptions extends BeanMappingOptions {
+
+        protected WrappedBeanMappingOptions(BeanMappingOptions options) {
+            super( options );
+        }
+
+        @Override
+        public BuilderGem getBuilder() {
+            return null;
```

Comment:
We should use the delegating here. Since builder can be defined on `@Mapper`. We only need to skip the `beanMapping`.

Reply:
done.. but I really need to debug this one..

Reply:
let's park this one for the time being..